### ISSUE TITLE
Expanding API

### ISSUE THREAD

**Issue by neomantic-zz** [asok/projectile-rails#33]
Projectile-rails is fantastic!  Bug free, in my experience, and with solid tests.  

I would like to build off it, but many of its procedure require `(interative)`.  For instance, I would love a non-interactive version of this procedure: https://github.com/asok/projectile-rails/blob/master/projectile-rails.el#L288, to simply get a list of rails "resources"

The only way I can do it now is something like this

``` elisp
(projectile-rails-choices '(("app/controllers/" "/controllers/\\(.+\\)_controller\\.rb$")))))
```

Basically, I just cut-and-paste from you code.  It would be cool, if I didn't have to duplicate such expressions.

Thoughts?
